morpho5/datastructures/sparse.c

```diff
@@ -811,11 +811,7 @@ value Sparse_setindex(vm *v, int nargs, value *args) {
     unsigned int indx[2]={0,0};
     
     if (array_valuestoindices(nargs-1, args+1, indx)) {
-        double val=0.0;
-        if (MORPHO_ISFLOAT(args[nargs])) val=MORPHO_GETFLOATVALUE(args[nargs]);
-        if (MORPHO_ISINTEGER(args[nargs])) val=(double) MORPHO_GETINTEGERVALUE(args[nargs]);
-
-        if (!sparse_setelement(s, indx[0], indx[1], MORPHO_FLOAT(val))) {
+        if (!sparse_setelement(s, indx[0], indx[1], args[nargs])) {
```

Comment:
I assume the float casting is now being done inside sparse_setelement?

Reply:
So the idea is that Sparse is no longer restricted strictly to numerical types. SparseDOK can in fact hold arbitrary values. There's more to do here... but it's a very useful change.